File: src/dataset1/generator.py
import random
from helper import write_file

def generate_data(size, digits):
    """Generate data set of a given size using the specified digits"""
    return [int("".join(random.choice(digits) for _ in range(3)))
        for _ in range(size)]

def main():
    # Allowed digits based on leader's ID
    allowed_digits = [int(d) for d in '1211310047']

    # Sizes for each dataset
    sizes = [100, 1_000, 10_000, 100_000, 500_000, 1_000_000]

    # Save Generated datasets to txt files
    for i, size in enumerate(sizes):
        data = generate_data(size, allowed_digits)
        print(data)

        file_name = f"sets/{i+1}.txt"
        write_file(data, file_name)
        print(f"Data for Set {i+1} saved to {file_name}")

if __name__ == '__main__':
    main()

File: src/Generator.js
const fs = require('fs');
const { join } = require('path');

function getRandomInt(max) {
    return Math.floor(Math.random() * max);
}

function generateData(size, digits) {
    // Generate data set of a given size using the specified digits
    let data = [];
    for (let i = 0; i < size; i++) {
        let numStr = '';
        for (let j = 0; j < 3; j++) {
            numStr += digits[getRandomInt(digits.length)];
        }
        data.push(parseInt(numStr, 10));
    }
    return data;
}

function writeFile(data, fileName) {
    fs.writeFile(fileName, data.join('\n'), (err) => {
        if (err) throw err;
        console.log(`Data for ${fileName} saved`);
    });
}

function main() {
    // Allowed digits based on leader's ID
    let allowedDigits = '1211310047'.split('').map(Number);

    // Sizes for each dataset
    let sizes = [100, 1000, 10000, 100000, 500000, 1000000];

    // Save generated datasets to txt files
    for (let i = 0; i < sizes.length; i++) {
        let size = sizes[i];
        let data = generateData(size, allowedDigits);
        console.log(data);

        let fileName = join(__dirname, `sets/${i + 1}.txt`);
        writeFile(data, fileName);
        console.log(`Data for Set ${i + 1} saved to ${fileName}`);
    }
}

main();